server/src/main/java/org/elasticsearch/index/mapper/RuntimeFieldType.java

```diff
@@ -8,179 +8,56 @@
 
 package org.elasticsearch.index.mapper;
 
-import org.apache.lucene.analysis.TokenStream;
-import org.apache.lucene.search.MultiTermQuery;
-import org.apache.lucene.search.Query;
-import org.apache.lucene.search.spans.SpanMultiTermQueryWrapper;
-import org.apache.lucene.search.spans.SpanQuery;
-import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.common.CheckedBiConsumer;
-import org.elasticsearch.common.geo.ShapeRelation;
-import org.elasticsearch.common.time.DateMathParser;
-import org.elasticsearch.common.unit.Fuzziness;
 import org.elasticsearch.common.xcontent.ToXContentFragment;
 import org.elasticsearch.common.xcontent.XContentBuilder;
-import org.elasticsearch.index.query.SearchExecutionContext;
 
 import java.io.IOException;
-import java.time.ZoneId;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;
-import java.util.Locale;
 import java.util.Map;
 import java.util.function.BiFunction;
-
-import static org.elasticsearch.search.SearchService.ALLOW_EXPENSIVE_QUERIES;
+import java.util.function.Function;
 
 /**
  * Base implementation for a runtime field that can be defined as part of the runtime section of the index mappings
  */
-public abstract class RuntimeFieldType extends MappedFieldType implements ToXContentFragment {
-
-    private final CheckedBiConsumer<XContentBuilder, Boolean, IOException> toXContent;
+public interface RuntimeFieldType extends ToXContentFragment {
 
-    protected RuntimeFieldType(String name, RuntimeFieldType.Builder builder) {
-        this(name, builder.meta(), builder::toXContent);
+    static <T> Function<FieldMapper, T> initializerNotSupported() {
+        return mapper -> { throw new UnsupportedOperationException(); };
     }
 
-    protected RuntimeFieldType(String name, Map<String, String> meta, CheckedBiConsumer<XContentBuilder, Boolean, IOException> toXContent) {
-        super(name, false, false, false, TextSearchInfo.SIMPLE_MATCH_WITHOUT_TERMS, meta);
-        this.toXContent = toXContent;
-    }
+    MappedFieldType asMappedFieldType(Function<String, MappedFieldType> lookup);
```

Comment:
with this, the fact that we have aliases kind of leaks into the runtime field base class. Maybe it is the way to make it explicit and avoid casting like we have for the existing alias field mapper, but I was hoping that the implementation of aliases can be more transparent and not require this.